pysb/core.py

```diff
@@ -1444,7 +1460,7 @@ def enable_synth_deg(self):
         source_cp = as_complex_pattern(self.monomers['__source']())
         if self.compartments:
             for c in self.compartments:
-                source_cp = source_cp ** c
+                source_cp.compartment = c
```

Comment:
You're right, there's an error. `source_cp` gets added to initials assigned to the first compartment in the `for` loop, then its compartment gets updated **in place**, such that `is_equivalent_to` returns true on subsequent loop iterations (since the patterns are the same object) and the pattern doesn't get added again. The end result is that we only have one `source()` which is assigned to the last compartment in the `for` loop due to the in place compartment update.

I'll write a patch using `copy()` and push it here.